software/py/bsg_manycore_link_gen.py

```diff
@@ -135,8 +135,9 @@ def script(self):
     # LMA (Load Memory Address)    => NPA used by loader
     # VMA (Virtual Memory Address) => Logical address used by linker for
     #                                 symbol resolutions
-    _DMEM_VMA_START   = 0x1000
-    _DMEM_VMA_SIZE    = 0x1000
+    # bsg-tommy: we reserve the first two words of DMEM for interrupt handlers.
+    _DMEM_VMA_START   = 0x0008
+    _DMEM_VMA_SIZE    = 0x0ff8
```

Comment:
This seems odd. The Tile can still access the addresses 0x0-0x7 right? Wouldn't that mean that the VMA region actually starts at 0x0000?

It seems like a cleaner way to do this would be to define VMA start to 0x0000, and then create variables at 0x0 and 0x4, either here, or in some assembly file. An added benefit would be that then C/C++ can directly dereference those locations instead of doing pointer manipulations

Am I missing something?

Reply:
This doesn't prevent accessing 0~7. The point is that we prevent the linker from placing objects in this space without having to include some special code.

Reply:
But why not declare symbols for the linker, and force it to reserve those spaces? That would make it easier to access the two reserved locations.

Reply:
okay, how do you do that?

Reply:
Actually, the location is already defined as nullptr. So the programmer can dereference from that?